### PR TITLE
feat: creating dateRangePicker

### PR DESCRIPTION
**Pull request by victorstassun**
### Description

Creating the new Date Range Picker component.

### Extra info

Passed design Q.A!!! 👍 

### Screenshots

Closed Input: 

image

Any date selected:

image

Only one date selected:

image

Range date selected:

image

### CODE REVIEW

File: src/components/Inputs/DateRangePicker/DateRangePicker.stories.mdx
Comment (gabrielforster) [inline src/components/Inputs/DateRangePicker/DateRangePicker.stories.mdx:23]:
```suggestion
      defaultValue: 'background-light',
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (gabrielforster) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:57]: computed for a cleaner code

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (gabrielforster) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:97]: computed?

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (gabrielforster) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:165]: computed?

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (gabrielforster) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:205]: computed?

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (gabrielforster) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:478]:
```suggestion
      const days = [...Array(daysInMonth).keys()].map(day => day + 1);
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (gabrielforster) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:532]:
```suggestion
      const { startDate, endDate } = this.state.inputValue;
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (gabrielforster) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:559]: Wouldn't be cool to use `decreaseDate` from adapcon-utils-js?
Is a simple function that receives a date and a certain duration do be decreased. I think its better than having numbers like this on the code

**Review by gabrielforster — CHANGES_REQUESTED** [review]
Small changes for better code!

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (victorstassun) [reply src/components/Inputs/DateRangePicker/DateRangePicker.vue:559]: Of course, I didn´t know about it, but I will change it.

**Review by caroltrindade — CHANGES_REQUESTED** [review]

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (gabrielforster) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:399]:
```suggestion
      return abreviatedMonth.slice(0, 1).toUpperCase() + abreviatedMonth.slice(1);
```

**Review by gabrielforster — APPROVED** [review]
LGTM

**Review by gabrielforster — DISMISSED** [review]
LGTM

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (caroltrindade) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:520]: image
The date format shouldn't be DD/MM/YYYY? Like 03/04/2023

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (caroltrindade) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:10]:
```suggestion
        @click="toggleDatePicker"
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (caroltrindade) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:239]:
```suggestion
              <p class="pb-md">{{ selectedPeriod() }}</p>
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (caroltrindade) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:54]:
```suggestion
                      @click="toggleMonthsSelector('startDate')"
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (caroltrindade) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:90]:
```suggestion
                    @click="toggleYearsSelector('startDate')"
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (caroltrindade) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:254]:
```suggestion
                @click.native="apply()"
```
or
```suggestion
                @click.native="saveDates()"
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (caroltrindade) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:405]:
```suggestion
    getSelectorClass(selectorType) {
      const bySelector = { month: this.state.isMonthSelectorOpen, year: this.state.isYearSelectorOpen }
      return bySelector[selectorType].startDate ? 'input input-opened' : 'input';
    },
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (caroltrindade) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:307]:
```suggestion
        monthSelector: {
          open: false,
          startDate: '',
          endDate: '',
        },
```

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (caroltrindade) [inline src/components/Inputs/DateRangePicker/DateRangePicker.vue:312]:
```suggestion
        isYearSelectorOpen: {
          open: false,
          startDate: '',
          endDate: '',
        },
```

**Review by caroltrindade — CHANGES_REQUESTED** [review]

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (gabrielforster) [reply src/components/Inputs/DateRangePicker/DateRangePicker.vue:521]: Changing it so the date is formatted via toLocaleDateString.

**Review by CauaKath — APPROVED** [review]
LGTM

**Review by gabrielforster — APPROVED** [review]
LGTM

**Review by mickaelnn — DISMISSED** [review]
LGTM

**Review by gabrielforster — COMMENTED** [review]
Some comments and suggestions.
This comments does not block the pr if it got 2 LGTM

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (victorstassun) [reply src/components/Inputs/DateRangePicker/DateRangePicker.vue:585]: I think the comment might be helpful, but I don´t see any reasons to make all this calc.

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (victorstassun) [reply src/components/Inputs/DateRangePicker/DateRangePicker.vue:372]: Did it, just changed the return of this function and removed the inputStyle object, now we return the hole style, but with a validation on the background-color

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (victorstassun) [reply src/components/Inputs/DateRangePicker/DateRangePicker.vue]: Already removed that blank space

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (gabrielforster) [reply src/components/Inputs/DateRangePicker/DateRangePicker.vue:569]: thats ok

File: src/components/Inputs/DateRangePicker/DateRangePicker.vue
Comment (gabrielforster) [reply src/components/Inputs/DateRangePicker/DateRangePicker.vue]: 🚀

**Review by gabrielforster — APPROVED** [review]
LGTM

**Review by mickaelnn — APPROVED** [review]
LGTM

**Pull request merged**
